docs/source/user.rst

```diff
@@ -0,0 +1,199 @@
+..  _user:
```

Comment:
On thing to think about is how to split this between the `README.md` and the documentation. I guess it makes sense to keep a few "appetizers" in the README but we should try to avoid code duplication wherever possible.